### PR TITLE
Create Github Actions workflow: Linting

### PR DESCRIPTION
**Pull request by erin-at-work**
Sets up Github Actions workflow for linting (formatting/eslint/type check)
- also fix type errors
image

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Checkout node action
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: ".nvmrc"
+
+      - name: Install NPM dependencies
+        run: yarn install --frozen-lockfile
+      
+      # - name: Runs compile asset script
+      #   run: ./compile-assets.js # FIXME: Permission denied error
```
Comment (erin-at-work) [inline .github/workflows/main.yml:32]: @ljharb How do I get past the permission error?

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+14.17
```
Comment (erin-at-work) [inline .nvmrc:1]: What's version should we be starting with?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: CI
+
+on:
+  push:
```
Comment (erin-at-work) [inline .github/workflows/main.yml:4]: Thinking updating this to run on `pull_request`

File: .github/workflows/main.yml (hunk 1 above)
Comment (ljharb) [reply .github/workflows/main.yml:32]: either `chmod a+x compile-assets.js` and commit the result, or, `node ./compile-assets`

File: .github/workflows/main.yml (hunk 3 above)
Comment (bangtoven) [reply .github/workflows/main.yml:4]: yes please

**Review by bangtoven — APPROVED** [review]

**Pull request merged**